magicctapipe/scripts/lst1_magic/semi_automatic_scripts/stereo_events.py

```diff
@@ -0,0 +1,302 @@
+"""
+This scripts generates and runs the bashscripts
+to compute the stereo parameters of DL1 MC and
+Coincident MAGIC+LST data files.
+
+Usage:
+$ python stereo_events.py (-c config.yaml)
+
+If you want to compute the stereo parameters only the real data or only the MC data,
+you can do as follows:
+
+Only real data:
+$ python stereo_events.py --analysis-type onlyReal (-c config.yaml)
+
+Only MC:
+$ python stereo_events.py --analysis-type onlyMC (-c config.yaml)
+"""
+
+import argparse
+import glob
+import logging
+import os
+from pathlib import Path
+
+import joblib
+import numpy as np
+import yaml
+
+from magicctapipe import __version__
+from magicctapipe.io import resource_file
+
+__all__ = ["configfile_stereo", "bash_stereo", "bash_stereoMC"]
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def configfile_stereo(ids, target_dir, source_name, NSB_match):
+
+    """
+    This function creates the configuration file needed for the event stereo step
+
+    Parameters
+    ----------
+    ids : list
+        List of telescope IDs
+    target_dir : str
+        Path to the working directory
+    source_name : str
+        Name of the target source
+    NSB_match : bool
+        If real data are matched to pre-processed MCs or not
+    """
+
+    config_file = resource_file("config.yaml")
+    with open(
+        config_file, "rb"
+    ) as fc:  # "rb" mode opens the file in binary format for reading
+        config_dict = yaml.safe_load(fc)
+    coincidence = config_dict["stereo_reco"]
+
+    conf = {}
+    conf["stereo_reco"] = coincidence
+
+    if not NSB_match:
+        file_name = f"{target_dir}/{source_name}/config_stereo.yaml"
+    else:
+        file_name = f"{target_dir}/v{__version__}/{source_name}/config_stereo.yaml"
+    with open(file_name, "w") as f:
+        lines = [
+            "mc_tel_ids:",
+            f"\n    LST-1: {ids[0]}",
+            f"\n    LST-2: {ids[1]}",
+            f"\n    LST-3: {ids[2]}",
+            f"\n    LST-4: {ids[3]}",
+            f"\n    MAGIC-I: {ids[4]}",
+            f"\n    MAGIC-II: {ids[5]}",
+            "\n",
+        ]
+        f.writelines(lines)
+        yaml.dump(conf, f, default_flow_style=False)
+
+
+def bash_stereo(target_dir, source, env_name, NSB_match):
```

Comment:
old comment: 
"pass the cluster type as one of the parameters"

Reply:
Done